internal/cluster/datasource_hagroup.go

```diff
@@ -0,0 +1,200 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+package cluster
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+
+	"github.com/bpg/terraform-provider-proxmox/internal/tffwk"
+	"github.com/bpg/terraform-provider-proxmox/proxmox"
+	hagroups "github.com/bpg/terraform-provider-proxmox/proxmox/cluster/ha/groups"
+)
+
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ datasource.DataSource              = &hagroupDatasource{}
+	_ datasource.DataSourceWithConfigure = &hagroupDatasource{}
+)
+
+// NewHAGroupDataSource is a helper function to simplify the provider implementation.
+func NewHAGroupDataSource() datasource.DataSource {
+	return &hagroupDatasource{}
+}
+
+// hagroupDatasource is the data source implementation for full information about
+// specific High Availability groups.
+type hagroupDatasource struct {
+	client *hagroups.Client
+}
+
+// hagroupModel maps the schema data for the High Availability group data source.
+type hagroupModel struct {
+	ID         types.String `tfsdk:"id"`
+	Group      types.String `tfsdk:"group"`
+	Comment    types.String `tfsdk:"comment"`
+	Members    types.Map    `tfsdk:"members"`
+	NoFailback types.Bool   `tfsdk:"no_failback"`
+	Restricted types.Bool   `tfsdk:"restricted"`
+}
+
+// Metadata returns the data source type name.
+func (d *hagroupDatasource) Metadata(
+	_ context.Context,
+	req datasource.MetadataRequest,
+	resp *datasource.MetadataResponse,
+) {
+	resp.TypeName = req.ProviderTypeName + "_hagroup"
+}
+
+// Schema returns the schema for the data source.
+func (d *hagroupDatasource) Schema(_ context.Context, _ datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "Retrieves information about a specific High Availability group.",
+		Attributes: map[string]schema.Attribute{
+			"id": tffwk.IDAttribute(),
+			"group": schema.StringAttribute{
+				Description: "The identifier of the High Availability group to read.",
+				Required:    true,
+			},
+			"comment": schema.StringAttribute{
+				Description: "The comment associated with this group",
+				Computed:    true,
+			},
+			"members": schema.MapAttribute{
+				Description: "The member nodes for this group, associated with their priority or to null if no priority is set.",
```

Comment:
I think the description is a little bit confusing, it's not immediately clear to me that `null` here is referencing to a "no value mapped to the group member name", not the `members` attribute itself.
I would also rename `members` -> `nodes` to match PVE API / UI naming

Reply:
I have no idea why I named it `members`, I was sure it was displayed as such in the UI but nope. I'll fix that, and the description too.